contracts/carrot-app/src/msg.rs

```diff
@@ -13,52 +18,76 @@ abstract_app::app_msg_types!(App, AppExecuteMsg, AppQueryMsg);
 /// App instantiate message
 #[cosmwasm_schema::cw_serde]
 pub struct AppInstantiateMsg {
-    /// Id of the pool used to get rewards
-    pub pool_id: u64,
-    /// Seconds to wait before autocompound is incentivized.
-    pub autocompound_cooldown_seconds: Uint64,
-    /// Configuration of rewards to the address who helped to execute autocompound
-    pub autocompound_rewards_config: AutocompoundRewardsConfig,
+    /// Future app configuration
+    pub config: ConfigUnchecked,
     /// Create position with instantiation.
     /// Will not create position if omitted
-    pub create_position: Option<CreatePositionMessage>,
-}
-
-#[cosmwasm_schema::cw_serde]
-pub struct CreatePositionMessage {
-    pub lower_tick: i64,
-    pub upper_tick: i64,
-    // Funds to use to deposit on the account
-    pub funds: Vec<Coin>,
-    /// The two next fields indicate the token0/token1 ratio we want to deposit inside the current ticks
-    pub asset0: Coin,
-    pub asset1: Coin,
-    // Slippage
-    pub max_spread: Option<Decimal>,
-    pub belief_price0: Option<Decimal>,
-    pub belief_price1: Option<Decimal>,
+    pub deposit: Option<Vec<Coin>>,
 }
 
 /// App execute messages
 #[cosmwasm_schema::cw_serde]
 #[cfg_attr(feature = "interface", derive(cw_orch::ExecuteFns))]
 #[cfg_attr(feature = "interface", impl_into(ExecuteMsg))]
 pub enum AppExecuteMsg {
-    /// Create the initial liquidity position
-    CreatePosition(CreatePositionMessage),
     /// Deposit funds onto the app
+    /// Those funds will be distributed between yield sources according to the current strategy
+    /// TODO : for now only send stable coins that have the same value as USD
+    /// More tokens can be included when the oracle adapter is live
     Deposit {
         funds: Vec<Coin>,
-        max_spread: Option<Decimal>,
-        belief_price0: Option<Decimal>,
-        belief_price1: Option<Decimal>,
+        /// This is additional paramters used to change the funds repartition when doing an additional deposit
+        /// This is not used for a first deposit into a strategy that hasn't changed for instance
+        /// This is an options because this is not mandatory
+        /// The vector then has option inside of it because we might not want to change parameters for all strategies
+        /// We might not use a vector but use a (usize, Vec<AssetShare>) instead to avoid having to pass a full vector everytime
+        yield_sources_params: Option<Vec<Option<Vec<AssetShare>>>>,
```

Comment:
It's a Vec because  if it's specified, one element needs to exists for each element in the underlying strategy. I'm not proud of the way params are passed to this function but I don't have identifiers for the strategy elements !